include/range/v3/view/linear_distribute.hpp

```diff
@@ -0,0 +1,115 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Eric Niebler 2013-2014
+//  Copyright Gonzalo Brito Gadeschi 2017
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+
+#ifndef RANGES_V3_VIEW_LINEAR_DISTRIBUTE_HPP
+#define RANGES_V3_VIEW_LINEAR_DISTRIBUTE_HPP
+#include <cmath>
+#include <limits>
+#include <type_traits>
+#include <range/v3/range_fwd.hpp>
+#include <range/v3/view_facade.hpp>
+#include <range/v3/utility/static_const.hpp>
+
+namespace ranges
+{
+    inline namespace v3
+    {
+        namespace view
+        {
+            /// \addtogroup group-views
+            /// @{
+
+            template <typename T>
+            struct linear_distribute_view
+              : view_facade<linear_distribute_view<T>, finite>
+            {
+                static_assert(std::is_arithmetic<T>(), "");
```

Comment:
Please `CONCEPT_ASSERT(foo)` all of these `static_assert(foo, "")` declarations.